src/PIL/Jpeg2KImagePlugin.py

```diff
@@ -131,11 +131,8 @@ def _res_to_dpi(num, denom, exp):
     """Convert JPEG2000's (numerator, denominator, exponent-base-10) resolution,
     calculated as (num / denom) * 10^exp and stored in dots per meter,
     to floating-point dots per inch."""
-    if num == 0 or denom == 0:
-        raise SyntaxError(
-            f"Invalid JP2 resolution information: ({num} / {denom}) * 10^{exp}"
-        )
-    return (254 * num * (10 ** exp)) / (10000 * denom)
+    if denom != 0:
+        return num / denom * (10 ** exp) * 0.0254
```

Comment:
I should've probably clarified this in a code comment: the reason for my roundabout way of calculating the resolution was to remain precise by working with ints for as long as possible, and to only have a single (division) operation which introduces floating point errors.

While your code is more is more elegant and more easily readable, there are now 3 operations that can introduce slight numeric errors.

I'm not sure whether my approach is overkill, though. It's not like those last few mantissa bits are ever realistically going to matter.

Reply:
Ah, ok. I was thinking of being consistent with how this value is used [elsewhere](https://github.com/python-pillow/Pillow/blob/9f42a17b2d03631a8131ae0f7ebc27bed6309482/src/PIL/PngImagePlugin.py#L503) in Pillow.

I'll switch back to your version.

Reply:
Awesome, thanks! 😄